### ISSUE TITLE
STG Error on startup, all tab changes during previous session lost

### ISSUE THREAD

**Issue by RevanProdigalKnight** [Drive4ik/simple-tab-groups#231]
**Describe the bug**
When re-starting Firefox, an error notification pops up, and no tab changes are recorded during the session, causing the same tabs to be re-opened the next time Firefox is restarted. Additionally, the STG extension menu does not populate in the top bar.

**To Reproduce**
Restart Firefox

**Expected behavior**
The extension does not pop up an error notification, tab changes are recorded during session, and STG menu appears in top bar

**Screenshots**
There's a content-policy error preventing me from uploading a screenshot, so here's the text of the error message. The notification cuts the entire stack trace off, and clicking the notification makes it disappear:

```
Error: [Exception... "Failure"  nsresult: "0x80004005 (NS_ERROR_FAILURE)"  location: "JS frame :: chrome://extensions/content/child/ext-storage.js :: fireOnChanged :: line 145"  data: no]
resource://g...
```

**Desktop (please complete the following information):**
 - OS: Windows 10 Professional (Build 17134.228)
 - Firefox Version: 63.0a1 (2018-08-30)

**Comment by RevanProdigalKnight** [comment]
After restoring the session and restarting the addon, it immediately reverted my tabs to the old session, and the menu still doesn't appear.

I'm starting to think that I may need to just uninstall and reinstall STG.

**Comment by RevanProdigalKnight** [comment]
Closed by accident

**Comment by Drive4ik** [comment]
Have you looked here?
![screenshot_1](https://user-images.githubusercontent.com/7843031/44850709-bc43c600-ac66-11e8-8c0b-f58ad1f0a747.png)

**Comment by RevanProdigalKnight** [comment]
![image](https://user-images.githubusercontent.com/5728037/44850878-dad09f80-ac1b-11e8-8504-2caa93c51b30.png)

**Comment by Drive4ik** [comment]
Ok. Try to turn off the addon, restart the browser and then turn it on

**Comment by RevanProdigalKnight** [comment]
The same thing happened as when simply disabling and re-enabling the extension.

**Comment by Drive4ik** [comment]
Ok. Have you tried to see what lies in the addon data folder?
1. about:support
1. click on "Open profile folder" button
1. go to `browser-extension-data\simple-tab-groups@drive4ik`

**Comment by RevanProdigalKnight** [comment]
The only thing in there is a file named `storage.js.migrated`

**Comment by Drive4ik** [comment]
Ok. Please open this file in notepad and if it has content like this:
```json
{
    "version": "3.1",
    "groups": [
        {
            "id": 1,
            "title": "Group 1",
            "tabs": [
                {
                    "title": "my title",
                    "url": "https://url.com",
                    "active": true
                }
            ],
            "iconColor": "rgb(0, 30, 255)"
        },
        {
            "id": 2,
            "title": "Group 2",
            "iconColor": "rgba(186, 186, 186, 0.8)",
            "tabs": [
                {
                    "title": "title 2",
                    "url": "https://test.com",
                    "active": false
                }
            ],
        }
    ],
    "createThumbnailsForTabs": true,
    "defaultGroupIconViewType": "squares",
    "hotkeys": [
        {
            "ctrlKey": true,
            "shiftKey": false,
            "altKey": false,
            "key": "`",
            "keyCode": 192,
            "action": "load-next-group",
            "metaKey": false
        }
    ]
}
```

then close browser and rename this file to `storage.js`

**Comment by RevanProdigalKnight** [comment]
The version in mine was `3.0.10`, and isn't formatted as nicely, but I closed my browser, renamed it, and started again. The error persists.

**Comment by Drive4ik** [comment]
Ok, let's try differently

1. copy folder `simple-tab-groups@drive4ik` to another location
1. remove addon
1. close browser
1. delete this folder (`simple-tab-groups@drive4ik`) if it is exist in the folder with the profile (`browser-extension-data`)
1. open the browser
1. install addon
1. if it works, even just open addon popup it well
1. close browser
1. copy the `simple-tab-groups@drive4ik` from location that you saved in step 1 into profile dir `browser-extension-data`
1. open the browser and wait to load tabs etc..

**Comment by RevanProdigalKnight** [comment]
It seems to be reading old extension data from somewhere else as soon as I reinstall it, and immediately reverts to my old tab session again.

**Comment by Drive4ik** [comment]
You reinstalled the add-on by this scheme: `remove addon/restart browser/install addon` ?

**Comment by RevanProdigalKnight** [comment]
Yes, with the additional step of deleting the extension data folder in between stopping and starting the browser.

**Comment by M7meDMost** [comment]
i had the same error 
it may be problem in file size
i used python code to delete "favIconUrl" and it's image of tabs in Base64 Encode 
so let's say the backup file is "stg-backup-2019-05-10@drive4ik.json",so the code will be
`
import json

with open('stg-backup-2019-05-10@drive4ik.json') as data_file:
    data = json.load(data_file)

for element in data:
    groups = data['groups']
    for item in groups:
        tabs = item['tabs']
        for tab in tabs:
            tab['favIconUrl'] = ''

with open('data.json', 'w') as outfile:
    json.dump(data, outfile)`

then you can restore from `data.json`
if didn't work remove STG then reinstall and final restore from `data.json`
